presto-docs/src/main/sphinx/installation/jmx.rst

```diff
@@ -0,0 +1,67 @@
+===================
+Monitoring with JMX
+===================
+
+Presto exposes a large number of different metrics via the Java Management
+Extensions (JMX). JConsole, supplied with the JDK, VisualVM, and many other
+tools can be used to access the metrics in a client application. Many monitoring
+solutions support JMX.
+
+You can configure the ports used by JMX in ``jvm.config`` as described in
+:doc:`/installation/deployment`.
+
+Alternatively you can use the :doc:`/connector/jmx` and query the metrics using
+SQL.
+
+Many of these JMX metrics are a complex metric object such as a ``CounterStat``
+that has a collection of related metrics. For example, ``InputPositions`` has
+``InputPositions.TotalCount``, ``InputPositions.OneMinute.Count``, and so on.
+
+The metrics can be grouped into a number of topics and the following sections
+describe a subset of all available metrics.
+
+JVM
+---
+
+* Heap size: ``java.lang:type=Memory:HeapMemoryUsage.used``
+* Thread count: ``java.lang:type=Threading:ThreadCount``
+
+Presto Cluster and Nodes
+------------------------
+
+* Active nodes
+``presto.failureDetector:name=HeartbeatFailureDetector:ActiveCount``
+
+* Free memory (general pool):
+``presto.memory:type=ClusterMemoryPool:name=general:FreeDistributedBytes``
+
+* Cumulative count of queries that ran out of memory and where killed, since
+Presto start:
+``presto.memory:name=ClusterMemoryManager:QueriesKilledDueToOutOfMemory``
+
+Presto Queries
+--------------
+
+* Active queries currently executing or queued: ``presto.execution:name=QueryManager:RunningQueries``
+
+* Queries started: ``presto.execution:name=QueryManager:StartedQueries.FiveMinute.Count``
+
+* Failed queries from last 5 min (all): ``presto.execution:name=QueryManager:FailedQueries.FiveMinute.Count``
+* Failed queries from last 5 min (internal): ``presto.execution:name=QueryManager:InternalFailures.FiveMinute.Count``
+* Failed queries from last 5 min (external): ``presto.execution:name=QueryManager:ExternalFailures.FiveMinute.Count``
+* Failed queries (user): ``presto.execution:name=QueryManager:UserErrorFailures.FiveMinute.Count``
+
+* Execution latency (P50): ``presto.execution:name=QueryManager:ExecutionTime.FiveMinutes.P50``
+* Input data rate (P90): ``presto.execution:name=QueryManager:WallInputBytesRate.FiveMinutes.P90``
+
+Presto Tasks
+------------
+
+* Input data bytes: ``presto.execution:name=TaskManager:InputDataSize.FiveMinute.Count``
+* Input rows: ``presto.execution:name=TaskManager:InputPositions.FiveMinute.Count``
+
+Hive Connector
```

Comment:
Let's remove this. We avoid having connector specific documentation outside of that connector's documentation. We could change this to
```
Connectors
----------

Many connectors provide their own metrics. These are typically named starting with ``presto.plugin``.
```
Ideally, we would document these metrics in the connector documentation. Then we could add a sentence above
> See connector documentation for details on available metrics.

Reply:
Applied with minor mod. Adding JMX docs to connectors is definitely something we should do going forward. Not in this PR though ..